### ISSUE TITLE
amplify update function command is no longer updating storage access permissions

### ISSUE THREAD

**Issue by rdutt** [aws-amplify/amplify-cli#10623]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm install -g @aws-amplify/cli

### If applicable, what version of Node.js are you using?

v16.15.1

### Amplify CLI Version

8.5.1

### What operating system are you using?

Amazon Linux 2 on Cloud9

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

Did not make any manual changes. 

### Amplify Categories

auth, storage, function, api, hosting

### Amplify Commands

update

### Describe the bug

Currently, the **amplify update function** command is broken after selecting the option to update resource access permissions, and selecting the storage option. Earlier this year, this was working correctly, and you were prompted to enter the name of the DynamoDB table that your Lambda function needed to access. Today, there is no option to enter the name of the table, and you are left with a Lambda function that cannot read the table.

To reproduce the issue:

$ amplify update function
? Select the Lambda function you want to update: my_function
? Which setting do you want to update? Resource access permissions
? Select the categories you want this function to have access to.
* storage

At this point, there should be a prompt to enter the name of the table, but that prompt is not given. This used to work correctly some months ago.

### Expected behavior

When the **storage** option is selected under the Resource Access Permissions option for the amplify update command, the user should be prompted for the name of the DynamoDB table that is being used for storage.  This was the behavior in the past, but is no longer the behavior today. As a result, the Lambda function never gets updated with the permissions required to access the DynamoDB table. 

### Reproduction steps

Follow all the steps given at https://aws.amazon.com/blogs/apn/how-to-build-a-fintech-app-on-aws-using-the-plaid-api/ 

The error happens during the following steps:

$ amplify update function
? Select the Lambda function you want to update: plaidaws
? Which setting do you want to update? Resource access permissions
? Select the categories you want this function to have access to.
* storage
At this point, the user is not prompted for the table name, so the Lambda function never gets the permissions it needs. 

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

**Comment by ykethan** [comment]
Hello @rdutt, Thank you for reaching out. From the information provided and steps utilized in the article. I understand that the application is adding 1 DynamoDB as storage resource. On `amplify update function` -> storage if there is only 1 storage resource available, amplify will select the resource by default in this case the DynamoDB table. On testing this in my Amplify application I was able to see this behavior as well.

![image](https://user-images.githubusercontent.com/87995712/174622381-041a9fed-ca4a-4b09-a742-7f773e047cb0.png)
as you can see amplify selected the DynamoDB by default. 

Additionally, on testing this with multiple DynamoDB tables the CLI recognized the tables as well. 
![image](https://user-images.githubusercontent.com/87995712/174622899-fff94d44-6947-4970-8dbf-f9dc8f81e6c8.png)
as you can see amplify recognized 2 resources

Could you please let us know if your amplify application has mutiple storage resources created and if there are any additional steps required to replicate the behaviour. Please feel free in adding screenshots for a better view into the issue as well.

**Comment by rdutt** [comment]
Hello @ykethan , thanks for the quick reply. Yes, my application uses only 1 storage resource. The easiest way to reproduce the issue is to start with an empty directory and follow the steps at this blog post - [How to Build a Fintech App on AWS](https://aws.amazon.com/blogs/apn/how-to-build-a-fintech-app-on-aws-using-the-plaid-api/) starting with the git clone step.   You do not need Plaid credentials, simply enter 1234 for the Plaid client ID and Plaid secret. 

After running the **amplify update function** command as described, check the Lambda execution role for the **plaidaws** function. It does not have the required policy to access the DynamoDB table.  Before the blog post was published in Jan 2022, the Lambda function _did_ have the DynamoDB policy added to it. 

Here's is the screen shot when I run the function:
![Screen Shot 2022-06-21 at 7 39 46 AM](https://user-images.githubusercontent.com/334888/174791087-70289bc1-4b3f-4366-87ac-cfc9513aa82b.png)

When I type <return> with the **storage** option selected, I get the following output:
![Screen Shot 2022-06-21 at 7 40 34 AM](https://user-images.githubusercontent.com/334888/174791254-e9e1cceb-2b50-4f2a-9c33-8e5057386aab.png)

This is different from the output you show on your screen shot. When I then look at the Lambda execution role of my plaidaws Lambda function, I do not see the DynamoDB policy added to it. 
![Screen Shot 2022-06-21 at 7 45 12 AM](https://user-images.githubusercontent.com/334888/174791785-4adc3007-0072-4896-b19c-72bbcbf1f753.png)

**Comment by rdutt** [comment]
@ykethan I just realized what the issue is.  To select the storage option, it is not sufficient to use the arrow key to get to the storage option. You have to hit the space bar also. If you don't hit the space bar, the option remains unselected. I think maybe the UI could be improved to let the user know that they have not selected any options. Thanks very much for your time. This issue may be closed.

**Comment by rdutt** [comment]
Yes, I was selecting using the enter key and not the space bar. This part of the UI is a bit confusing. Thank you very much for your help.
